compiler/rustc_ast_lowering/src/delegation.rs

```diff
@@ -273,26 +272,72 @@ impl<'hir> LoweringContext<'_, 'hir> {
         })
     }
 
-    // Generates fully qualified call for the resulting body.
+    fn lower_target_expr(&mut self, block: &Block) -> hir::Expr<'hir> {
```

Comment:
Could you add an short explanation comment here, with a link to https://github.com/rust-lang/rfcs/pull/3530#issuecomment-2197170600?